custom_components/enphase_envoy/envoy_endpoints.py

```diff
@@ -71,27 +71,27 @@
     },
     "devstatus": {
         "url": "https://{}/ivp/peb/devstatus",
-        "cache": 20,
+        "cache": 0,
         "installer_required": True,
         "optional": False,
     },
     "pcu_comm_status": {
         "url": "https://{}/installer/pcu_comm_check",
-        "cache": 90,
+        "cache": 3600,
         "installer_required": True,
         "optional": True,
     },
     # Netprofile endpoints
     "installer_agf": {
         "url": "https://{}/installer/agf/index.json",
-        "cache": 10,
+        "cache": 3600,
```

Comment:
It might be better to keep this as before (basically uncached). I use automation to switch grid profiles based on the current wholesale export price on the grid. Since the price changes very quickly (update every few minutes), the grid profile can also be switched very quickly. It would be good to monitor what the system thinks the current grid profile is as the grid profile switches.

I have been puzzled by that the grid profile sensor isn't updating as quickly as before after an update. I guess this is the culprit.

If we want to keep the traffic low, it could alternatively try to invalidate the cache when set grid profile action is invoked?

Reply:
We actually invalidate the cache already [here](https://github.com/vincentwolsink/home_assistant_enphase_envoy_installer/blob/d92c69c9bc0bc4a83db50da2702c5d49a1468256/custom_components/enphase_envoy/envoy_reader.py#L1271). But if there is a delay in the Envoy updating the value this might not work. Since this specific endpoint is pretty fast I can disable caching again.

Reply:
I think it delays by like a minute or two, so yeah probably invalidating the cache isn't enough.